tests/integration/cag/test_inequality.py

```diff
@@ -0,0 +1,606 @@
+        'tables': {
+            'table1': {
+                'columns': {
+                    'A': {'sdtype': 'numerical'},
+                    'B': {'sdtype': 'numerical'},
+                }
+            },
+            'table2': {
+                'columns': {
+                    'id': {'sdtype': 'id'},
+                }
+            },
+        }
+    })
+    pattern = Inequality(
+        low_column_name='A',
+        high_column_name='B',
+        table_name='table1',
+    )
+
+    # Run
+    pattern.validate(data, metadata)
+    updated_metadata = pattern.get_updated_metadata(metadata)
+    pattern.fit(data, metadata)
+    transformed = pattern.transform(data)
+    reverse_transformed = pattern.reverse_transform(transformed)
+
+    # Assert
+    expected_updated_metadata = Metadata.load_from_dict({
+        'tables': {
+            'table1': {
+                'columns': {
+                    'A': {'sdtype': 'numerical'},
+                    'A#B': {'sdtype': 'numerical'},
+                }
+            },
+            'table2': {
+                'columns': {
+                    'id': {'sdtype': 'id'},
+                }
+            },
+        }
+    }).to_dict()
+    assert expected_updated_metadata == updated_metadata.to_dict()
+    assert list(transformed['table1'].columns) == ['A', 'A#B']
+    assert set(data.keys()) == set(reverse_transformed.keys())
+    for table_name, table in data.items():
+        pd.testing.assert_frame_equal(table, reverse_transformed[table_name])
+
+
+@pytest.mark.skip(reason='Skipping until add_cag method is implemented')
+def test_inequality_with_timestamp_and_date():
+    """Test that the inequality pattern passes without strict boundaries.
+
+    This test checks if the `Inequality` pattern can handle two columns
+    with different datetime formats when `strict_boundaries` is set to `False`.
+    The pattern allows the `SUBMISSION_TIMESTAMP` column to be less than
+    or equal to the `DUE_DATE` column, even when they differ in precision but end
+    within the same day.
+    """
+    # Setup
+    data = pd.DataFrame(
+        data={
+            'SUBMISSION_TIMESTAMP': [
+                '2016-07-10 17:04:00',
+                '2016-07-11 13:23:00',
+                '2016-07-12 08:45:30',
+                '2016-07-11 12:00:00',
+                '2016-07-12 10:30:00',
+            ],
+            'DUE_DATE': ['2016-07-10', '2016-07-11', '2016-07-12', '2016-07-13', '2016-07-14'],
+        }
+    )
+
+    metadata = Metadata.load_from_dict({
+        'tables': {
+            'table': {
+                'columns': {
+                    'SUBMISSION_TIMESTAMP': {
+                        'sdtype': 'datetime',
+                        'datetime_format': '%Y-%m-%d %H:%M:%S',
+                    },
+                    'DUE_DATE': {'sdtype': 'datetime', 'datetime_format': '%Y-%m-%d'},
+                }
+            }
+        }
+    })
+    synthesizer = GaussianCopulaSynthesizer(metadata)
```

Comment:
All the tests with `GaussianCopulaSynthesizer` use datetime columns (for low/high column_name). Do we want a test with numerical columns?